common-design/sass/cd-header/_cd-search--inline.scss

```diff
@@ -0,0 +1,119 @@
+/**
+ * Common Design Search form in header.
+ */
+
+.cd-search--inline {
+  float: left;
+  border-left: 1px solid $cd-border-color;
+  border-right: 1px solid $cd-border-color;
+
+  .cd-search__form {
```

Comment:
Typically when I see BEM used in Sass the nesting is not present in the compiled CSS. The `&` selector can be used to nest SCSS but produce flat, single-class CSS. These single classes are often the goal of using a BEM convention in order to avoid extra specificity caused by nested classes. The overrides are easier because they only have to override one class. An example:

```scss
.cd-search {
  // CD search block

  &--inline {
    // CD Search--Inline variant
  }

  &__form {
    // CD Search: form element
  }

  &__form-inner {
    // CD Search: form-inner element
  }
}
```

Even though it's nested and DRY in the Sass source, the output CSS looks like this:

```css
.cd-search {}
.cd-search--inline {}
.cd-search__form {}
.cd-search__form-inner {}
```

When overrides need to be applied to any block, element, or variant, the specificity only has to overrule the one classname instead of discovering and overriding whatever nesting is present in the default styles.

Reply:
Ok. This was a first attempt to seeing if I could produce the two variants with CSS only, and as I think you pointed out, it's a bit sloppy :)
I have ah html file for search--inline so I can adjust all classes as needed.
I'd opt for adding proper classes to the markup, but I also wanted to see what classes and rules would be common between `search` and `search--inline`
Consider this a work in progress for now